node/core/av-store/src/lib.rs

```diff
@@ -19,52 +19,290 @@
 	Subsystem(SubsystemError),
+	#[from]
+	Time(SystemTimeError),
+}
+
+/// A key for chunk pruning records.
+const CHUNK_PRUNING_KEY: [u8; 14] = *b"chunks_pruning";
+
+/// A key for PoV pruning records.
+const POV_PRUNING_KEY: [u8; 11] = *b"pov_pruning";
+
+/// A key for a cached value of next scheduled PoV pruning.
+const NEXT_POV_PRUNING: [u8; 16] = *b"next_pov_pruning";
+
+/// A key for a cached value of next scheduled chunk pruning.
+const NEXT_CHUNK_PRUNING: [u8; 18] = *b"next_chunk_pruning";
+
+/// The following constants are used under normal conditions:
+
+/// Stored block is kept available for 1 hour.
+const KEEP_STORED_BLOCK_FOR: u64 = 60*60;
+
+/// Finalized block is kept for 1 day.
+const KEEP_FINALIZED_BLOCK_FOR: u64 = 24 * 60 * 60;
+
+/// Keep chunk of the finalized block for 1 day + 1 hour.
+const KEEP_FINALIZED_CHUNK_FOR: u64 = 25 * 60 * 60;
+
+/// At which point in time we need to wakeup and do next pruning of blocks.
+/// Essenially this is the first element in the sorted array of pruning data,
+/// we just want to cache it here to avoid lifting the whole array just to look at the head.
+///
+/// This record exists under `NEXT_POV_PRUNING` key, if it does not either:
+///  a) There are no records and nothing has to be pruned.
+///  b) There are records but all of them are in `Included` state and do not have exact time to
+///     be pruned.
+#[derive(Decode, Encode)]
+struct NextPoVPruning(u64);
+
+impl NextPoVPruning {
+	// After which amount of seconds into the future from `now` this should fire.
+	fn should_fire_in(&self) -> Result<u64, Error> {
+		Ok(self.0.saturating_sub(SystemTime::now().duration_since(UNIX_EPOCH)?.as_secs()))
+	}
+}
+
+/// At which point in time we need to wakeup and do next pruning of chunks.
+/// Essenially this is the first element in the sorted array of pruning data,
+/// we just want to cache it here to avoid lifting the whole array just to look at the head.
+///
+/// This record exists under `NEXT_CHUNK_PRUNING` key, if it does not either:
+///  a) There are no records and nothing has to be pruned.
+///  b) There are records but all of them are in `Included` state and do not have exact time to
+///     be pruned.
+#[derive(Decode, Encode)]
+struct NextChunkPruning(u64);
+
+impl NextChunkPruning {
+	// After which amount of seconds into the future from `now` this should fire.
+	fn should_fire_in(&self) -> Result<u64, Error> {
+		Ok(self.0.saturating_sub(SystemTime::now().duration_since(UNIX_EPOCH)?.as_secs()))
+	}
+}
+
+/// Struct holding pruning timing configuration.
+/// The only purpose of this structure is to use different timing
+/// configurations in production and in testing.
+#[derive(Clone)]
+struct PruningConfig {
+	/// How long should a stored block stay available.
+	keep_stored_block_for: u64,
+
+	/// How long should a finalized block stay available.
+	keep_finalized_block_for: u64,
+
+	/// How long should a chunk of a finalized block stay available.
+	keep_finalized_chunk_for: u64
```

Comment:
What are the units? Perhaps, it'd be better to change the naming to incorporate `_secs` to make it immediatelly clear.